lib/puppet/provider/katello_ssl_tool.rb

```diff
@@ -114,41 +59,20 @@ def common_args
        '--cert-expiration', resource[:expiration]]
     end
 
-    def rpmfile
-      path = self.build_path("#{rpmfile_base_name}")
-      path = path + "-[0-9].*" + "noarch.rpm"
-
-      rpmfile = Dir[path].max_by do |file|
-        version_from_name(file)
+    def pubkey(cert_name = resource[:name])
+      if resource.to_hash.key?(:hostname)
+        "#{resource[:build_dir]}/#{resource[:hostname]}/#{cert_name}.crt"
+      else
+        "#{resource[:build_dir]}/#{cert_name}.crt"
       end
-
-      rpmfile ||= self.build_path("#{rpmfile_base_name}.noarch.rpm")
-      return rpmfile
-    end
-
-    # file that indicates that a new version of the rpm should be updated
-    def update_file
-      self.build_path("#{rpmfile_base_name}.update")
     end
 
-    def rpmfile_base_name
-      resource[:name]
-    end
-
-    def pubkey
-      self.class.pubkey(resource[:name])
-    end
-
-    def privkey
-      self.class.privkey(resource[:name])
-    end
-
-    def target_path(file_name = '')
-      self.class.target_path(file_name)
-    end
-
-    def self.target_path(file_name = '')
-      File.join("/etc/pki/katello-certs-tools", file_name)
+    def privkey(key_name = resource[:name])
+      if resource.to_hash.key?(:hostname)
+        "#{resource[:build_dir]}/#{resource[:hostname]}/#{key_name}.key"
+      else
+        "#{resource[:build_dir]}/#{key_name}.key"
+      end
```

Comment:
shouldn't this be using `build_path` (which, ironically, doesn't use `:build_dir`)?